src/extension/MimeInspector/Utilities/MimeParser.cs

```diff
@@ -43,17 +43,44 @@ public static MimeMessage ParseMessage(Stream inputStream, HTTPHeaders headers,
                 return null;
             }
 
-            string headerString = String.Join("\r\n", headers.ToArray().Select(h => $"{h.Name}:{h.Value}"));
+            var headerArray = headers.ToArray();
 
-            using (ChainedStream streamWithHeaders = new ChainedStream())
+            if(IsMimeMessage(headerArray) == false)
             {
-                streamWithHeaders.Add(new MemoryStream(Encoding.UTF8.GetBytes(headerString)), false);
-                streamWithHeaders.Add(inputStream, false);
+                return null;
+            }
+
+            string headerString = String.Join("\r\n", headerArray.Select(h => $"{h.Name}:{h.Value}"));
```

Comment:
Should there be a space between header name and value?

Reply:
[RFC 7230](https://tools.ietf.org/html/rfc7230#section-3.2) states that the whitespace is optional.  But if you rather have a whitespace, I'll add one.  Just let me know

Reply:
Today I learned something new :). If it's not disallowed I would add it for readability

Reply:
done.